controllers/profiles/reconciler_dev.go

```diff
@@ -324,11 +363,114 @@ func mountDevConfigMapsMutateVisitor(flowDefCM, propsCM *v1.ConfigMap) mutateVis
 			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = make([]v1.VolumeMount, 0)
 			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
 
+			for keycm, cm := range externalResourceConfigMaps {
+				if !isContained(cm.Name, deployment.Spec.Template.Spec.Volumes) {
```

Comment:
Not sure why we need this conditional here. You can create the volume array and override it in the deployment at the end of the method.

Reply:
Isn't more needed